KnowYourEnemyMutagen/Program.cs

```diff
@@ -39,315 +29,219 @@ public static int Main(string[] args)
-                    {
-                        effect.BaseEffect.TryResolve(state.LinkCache, out var baseEffect);
-                        if (baseEffect != null && baseEffect.EditorID != null)
-                        {
-                            if (resistances_and_weaknesses.Contains(baseEffect.EditorID))
-                            {
-                                if (effect.Data != null)
-                                {
-                                    effect.Data.Magnitude = 0;
-                                }
-                                else
-                                {
-                                    Console.WriteLine("Error setting Effect Magnitude - DATA was null!");
-                                }
-                            }
-                        }
-                    }
-                    state.PatchMod.Spells.GetOrAddAsOverride(modifiedSpell);
+                    effect.BaseEffect.TryResolve(state.LinkCache, out var baseEffect);
+                    if (baseEffect?.EditorID == null) continue;
+                    if (!resistancesAndWeaknesses.Contains(baseEffect.EditorID)) continue;
+                    if (effect.Data != null)
+                        effect.Data.Magnitude = 0;
+                    else
+                        Console.WriteLine("Error setting Effect Magnitude - DATA was null!");
                 }
+
+                state.PatchMod.Spells.GetOrAddAsOverride(modifiedSpell);
             }
-            // ***** PART 1b *****
+
             // Remove other weapon resistance systems
             foreach (var perk in state.LoadOrder.PriorityOrder.WinningOverrides<IPerkGetter>())
             {
-                if (perk.EditorID != null && perks_to_clean.Contains(perk.EditorID))
+                if (perk.EditorID == null || !perksToClean.Contains(perk.EditorID)) continue;
+                foreach (var eff in perk.Effects)
                 {
-                    foreach (var eff in perk.Effects)
-                    {
-                        if (!(eff is PerkEntryPointModifyValue modValue)) continue;
-                        if (modValue.EntryPoint != APerkEntryPointEffect.EntryType.ModIncomingDamage) continue;
-                        modValue.Value = 1f;
-                        modValue.Modification = PerkEntryPointModifyValue.ModificationType.Set;
-                    }
+                    if (!(eff is PerkEntryPointModifyValue modValue)) continue;
+                    if (modValue.EntryPoint != APerkEntryPointEffect.EntryType.ModIncomingDamage) continue;
+                    modValue.Value = 1f;
+                    modValue.Modification = PerkEntryPointModifyValue.ModificationType.Set;
                 }
             }
 
-            // ***** PART 2a *****
             // Adjust KYE's physical effects according to effect_intensity
-            if (effect_intensity != 1)
+            if (Math.Abs(effectIntensity - 1) > float.Epsilon)
```

Comment:
Could make use of CSharpExt's EqualsWithin functionality: `!effect_intensity.EqualsWithin(1)` 
I'm assuming that's what this is doing?

Reply:
Yes, it checks whether the float is set to the default of 1 in the json config file